pandas/core/generic.py

```diff
@@ -1906,6 +1906,10 @@ def _slice(self, slobj, axis=0, kind=None):
         return result
 
     def _set_item(self, key, value):
```

Comment:
That is my understanding, yes. Here is an example of setting while using `.loc`:

```python
import pandas as pd
df = pd.DataFrame({'one': [0, 1, 2]})
df.loc[:, 'sum'] = df.one.sum()
```

```
UserWarning: Column name 'sum' collides with a built-in method, which will cause unexpected attribute behavior
  self._set_item(key, value)

```